### PR TITLE
feat: ability to init i18n with IntlConfig params overridden [KHCP-7670]

### PR DESCRIPTION
**Pull request by ValeryG**
# Summary

Details in:  https://kongstrong.slack.com/archives/C04A6LD904E/p1686302933215479

### CODE REVIEW

File: packages/core/i18n/README.md
Comment (johncowen) [inline packages/core/i18n/README.md:103]:
```suggestion
For some applications there is a need to create Intl instance with config settings overridden. One example of this is when application needs to define error/warning handlers. In this case use `createI18nEx` function.
```

Just a nit/typo I saw

File: packages/core/i18n/README.md
Comment (johncowen) [inline packages/core/i18n/README.md:117]:
```suggestion
    locale: 'en-us',
```

Just a little nit/typo

**Review by johncowen — APPROVED** [review]
Code here LGTM, guessing you'll want a proper approval from other folks on your team tho.

Just out of interest more than anything, how come the extra function? Back compat with previous versions or something?

Also I've not tried this in place myself, let me know if you want me to before you merge this tho (I just didn't try it as it looks pretty good to me from the sandbox and you've added a test for proof etc)

File: packages/core/i18n/src/i18n.ts
Comment (adamdehaven) [inline packages/core/i18n/src/i18n.ts:14]: suggestion: we need to add TS Doc formatted comment above each function here so that users can properly identity which one they want to import.

In my opinion, the two exported functions here are named too similarly. Could we change the name of the new function to something like `createI18nWithConfig` instead? (I think that is what it's actually doing, right?)
```suggestion
/**
 * {Description of this function}
 * @param {IntlConfigEx<MessageSource>} config {Description of config param}
 * @param {boolean} isGlobal {Description of isGlobal param)
 *
*/
export const createI18nEx = <MessageSource extends Record<string, any>>(config: IntlConfigEx<MessageSource>, isGlobal: boolean = false): IntlShapeEx<MessageSource> => {
```

**Comment by adamdehaven** [comment]
@ValeryG as an alternative to adding a new function, could we keep using the **existing** function, but add a third parameter of the config object that defaults to the default config? Then consuming apps wouldn’t have to change their usage, but you could optionally pass in a config object?

**Comment by adamdehaven** [comment]
We could do a breaking change (bumping the major version) to refactor how we pass `messages` and `locale` so that it's part of a config object, and then an app that wants to pass __more__ parameters can choose to do so.

Would this be the better solution so that we don't have two similar functions exported?

**Comment by ValeryG** [comment]
not a big fan of doing breaking changes - this is the case when we can avoid it ... :) here is what I am proposing instead (changes are committed in this PR): 
```ts
// still works , creates local instance (same as before)
createI18n<typeof english>('en-us', english) 

// still works , creates or returns global instance (same as before)
createI18n<typeof english>('en-us', english, true) 

// new way , creates or returns global instance, with possibility to define additional IntlConfig settings
createI18n<typeof english>('en-us', english, {isGLobal: true, onError: (err: any) {console.error(err)}} ) 
```

hope we like it :) 
```

**Review by johncowen — COMMENTED** [review]
Just had another quick look here as I saw things had changed from your link in Slack Val, thanks for that! This is fine by me, but guessing you want final approval from this team. Wanted to say thanks for doing the extra work here 👍 and just so you know I'd looked again

**Review by adamdehaven — APPROVED** [review]
The code LGTM; did not functionally test

**Pull request merged**
